### ISSUE TITLE
Downloads with `--check-all-formats` are broken

### ISSUE THREAD

**Issue by scolby33** [yt-dlp/yt-dlp#1829]
### Checklist

- [X] I'm reporting a bug unrelated to a specific site
- [X] I've verified that I'm running yt-dlp version **2021.11.10.1**. ([update instructions](https://github.com/yt-dlp/yt-dlp#update))
- [X] I've checked that all provided URLs are alive and playable in a browser
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/ytdl-org/youtube-dl#video-url-contains-an-ampersand-and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues including closed ones. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Description

YouTube downloads with `--check-all-formats` fail with `TypeError: cannot pickle 'generator' object`

The problem lies with the code here https://github.com/yt-dlp/yt-dlp/commit/9f1a1c36e60b14f9ff47d83234b4ea61c5f5e2f7#diff-d3ba8be45cae8dd7889a71c3360c9e4ac1160de8a5f3443b6e4a656395267f9bR2135-R2143, which produces a generator, later stored in the `info` dict and subsequently `copy.deepcopy()`'d here https://github.com/yt-dlp/yt-dlp/blob/master/yt_dlp/YoutubeDL.py#L2595.

### Verbose log

```shell
$ yt-dlp --verbose --check-all-formats 'https://www.youtube.com/watch?v=BaW_jenozKc'
[debug] Command-line config: ['--verbose', '--check-all-formats', 'https://www.youtube.com/watch?v=BaW_jenozKc']
[debug] User config file: /Users/scott/.config/yt-dlp/config
[debug] User config: ['--restrict-filenames', '--prefer-free-formats', '--check-formats', '--merge-output-format', 'mkv', '--sub-langs', 'all,-live_chat', '--embed-subs', '--embed-metadata', '--embed-chapters']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, err utf-8, pref UTF-8
[debug] yt-dlp version 2021.11.10.1
[debug] Python version 3.9.7 (CPython 64bit) - macOS-11.6-x86_64-i386-64bit
[debug] exe versions: ffmpeg 4.4.1 (setts), ffprobe 4.4.1, rtmpdump 2.4
[debug] Optional libraries: Cryptodome, mutagen, sqlite, websockets
[debug] Proxy map: {}
[debug] [youtube] Extracting URL: https://www.youtube.com/watch?v=BaW_jenozKc
[youtube] BaW_jenozKc: Downloading webpage
[youtube] BaW_jenozKc: Downloading android player API JSON
[debug] Sort order given by extractor: quality, res, fps, hdr:12, source, codec:vp9.2, lang, proto
[debug] Formats sorted by: hasvid, ie_pref, quality, res, fps, hdr:12(7), source, vcodec:vp9.2(10), acodec, lang, proto, filesize, fs_approx, tbr, vbr, abr, asr, vext, aext, hasaud, id
[debug] Downloading subtitles: en
[info] Testing format 248
[download] Destination: /var/folders/_d/b50dfp0j3vd4ybthlxdzmw740000gn/T/tmp4msxw87t.tmp
[download] 100% of 10.00KiB in 00:00
[info] Testing format 137
[download] Destination: /var/folders/_d/b50dfp0j3vd4ybthlxdzmw740000gn/T/tmptrc05psj.tmp
[download] 100% of 10.00KiB in 00:00
[info] Testing format 247
[download] Destination: /var/folders/_d/b50dfp0j3vd4ybthlxdzmw740000gn/T/tmpd8p4cdyw.tmp
[download] 100% of 10.00KiB in 00:00
[info] Testing format 22
[download] Destination: /var/folders/_d/b50dfp0j3vd4ybthlxdzmw740000gn/T/tmp4domvv01.tmp
[download] 100% of 10.00KiB in 00:00
[info] Testing format 136
[download] Destination: /var/folders/_d/b50dfp0j3vd4ybthlxdzmw740000gn/T/tmpfageafif.tmp
[download] 100% of 10.00KiB in 00:00
[info] Testing format 244
[download] Destination: /var/folders/_d/b50dfp0j3vd4ybthlxdzmw740000gn/T/tmp9tr9dohf.tmp
[download] 100% of 10.00KiB in 00:00
[info] Testing format 135
[download] Destination: /var/folders/_d/b50dfp0j3vd4ybthlxdzmw740000gn/T/tmped7frt6y.tmp
[download] 100% of 10.00KiB in 00:00
[info] Testing format 243
[download] Destination: /var/folders/_d/b50dfp0j3vd4ybthlxdzmw740000gn/T/tmpglf0wsyy.tmp
[download] 100% of 10.00KiB in 00:00
[info] Testing format 18
[download] Destination: /var/folders/_d/b50dfp0j3vd4ybthlxdzmw740000gn/T/tmpamp8kddo.tmp
[download] 100% of 10.00KiB in 00:00
[info] Testing format 134
[download] Destination: /var/folders/_d/b50dfp0j3vd4ybthlxdzmw740000gn/T/tmptngra_6i.tmp
[download] 100% of 10.00KiB in 00:00
[info] Testing format 242
[download] Destination: /var/folders/_d/b50dfp0j3vd4ybthlxdzmw740000gn/T/tmp8spclz4_.tmp
[download] 100% of 10.00KiB in 00:00
[info] Testing format 133
[download] Destination: /var/folders/_d/b50dfp0j3vd4ybthlxdzmw740000gn/T/tmpbkxytujd.tmp
[download] 100% of 10.00KiB in 00:00
[info] Testing format 278
[download] Destination: /var/folders/_d/b50dfp0j3vd4ybthlxdzmw740000gn/T/tmpbnu_6d_7.tmp
[download] 100% of 10.00KiB in 00:00
[info] Testing format 160
[download] Destination: /var/folders/_d/b50dfp0j3vd4ybthlxdzmw740000gn/T/tmpjfho_hm3.tmp
[download] 100% of 10.00KiB in 00:00
[info] Testing format 17
[download] Destination: /var/folders/_d/b50dfp0j3vd4ybthlxdzmw740000gn/T/tmp1edjxtvp.tmp
[download] 100% of 10.00KiB in 00:00
[info] Testing format 251
[download] Destination: /var/folders/_d/b50dfp0j3vd4ybthlxdzmw740000gn/T/tmplbab_l14.tmp
[download] 100% of 10.00KiB in 00:00
[info] Testing format 140
[download] Destination: /var/folders/_d/b50dfp0j3vd4ybthlxdzmw740000gn/T/tmp5rv_z2sd.tmp
[download] 100% of 10.00KiB in 00:00
[info] Testing format 250
[download] Destination: /var/folders/_d/b50dfp0j3vd4ybthlxdzmw740000gn/T/tmp8gqzv8jv.tmp
[download] 100% of 10.00KiB in 00:00
[info] Testing format 249
[download] Destination: /var/folders/_d/b50dfp0j3vd4ybthlxdzmw740000gn/T/tmp8ltfhsl1.tmp
[download] 100% of 10.00KiB in 00:00
[info] Testing format 139
[download] Destination: /var/folders/_d/b50dfp0j3vd4ybthlxdzmw740000gn/T/tmpg6vrnis_.tmp
[download] 100% of 10.00KiB in 00:00
[debug] Default format spec: bestvideo*+bestaudio/best
[info] BaW_jenozKc: Downloading 1 format(s): 248+251
[info] Writing video subtitles to: youtube-dl_test_video_a-[BaW_jenozKc].en.vtt
[debug] Invoking downloader on "https://www.youtube.com/api/timedtext?v=BaW_jenozKc&asr_langs=de%2Cen%2Ces%2Cfr%2Cid%2Cit%2Cja%2Cko%2Cnl%2Cpt%2Cru%2Ctr%2Cvi&caps=asr&exp=xftt%2Cxctw&xoaf=5&hl=en&ip=0.0.0.0&ipbits=0&expire=1638184345&sparams=ip%2Cipbits%2Cexpire%2Cv%2Casr_langs%2Ccaps%2Cexp%2Cxoaf&signature=7B74F9FB92D0A9C307C4F4CB45AD3992B534CA4E.AB34A87DB22671BFCADBB625B447F1416A6C38F2&key=yt8&lang=en&fmt=vtt"
[download] youtube-dl_test_video_a-[BaW_jenozKc].en.vtt has already been downloaded
[download] 100% of 177.00B[debug] Invoking downloader on "https://r3---sn-p5qlsndd.googlevideo.com/videoplayback?expire=1638180746&ei=KlOkYe_1DcL48gTx06ZQ&ip=2601%3A547%3A901%3A6850%3Aa47c%3A2851%3Adf53%3Ad7c8&id=o-AD5AIKezodK0qInqiTMdj4_2Dh3ECotszT4RkHMXbOQ5&itag=248&source=youtube&requiressl=yes&mh=Rf&mm=31%2C29&mn=sn-p5qlsndd%2Csn-p5qs7nee&ms=au%2Crdu&mv=m&mvi=3&pl=40&initcwndbps=1735000&vprv=1&mime=video%2Fwebm&gir=yes&clen=988479&dur=9.833&lmt=1544741703689523&mt=1638158943&fvip=3&keepalive=yes&fexp=24001373%2C24007246&beids=23886200&c=ANDROID&txp=5432432&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cvprv%2Cmime%2Cgir%2Cclen%2Cdur%2Clmt&sig=AOq0QJ8wRgIhALn4q9AFBiwBr0KSNjPwBxfMYhM9N5S8gknvv2LrTO09AiEA0hyUF9tTC4celJOIXCQtrgoPOgaZ-mNB5p5vOFVN7Fg%3D&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps&lsig=AG3C_xAwRQIgDBGSu1l81yz379WG881J_ekPYqLrrrvbsppbcj89fB4CIQD9rtM-LiyPinpQDC82hkmAKdbxWnCtdwQfrBBcmv1UnA%3D%3D"
ERROR: cannot pickle 'generator' object
Traceback (most recent call last):
  File "/Users/scott/.local/pipx/venvs/yt-dlp/lib/python3.9/site-packages/yt_dlp/YoutubeDL.py", line 1311, in wrapper
    return func(self, *args, **kwargs)
  File "/Users/scott/.local/pipx/venvs/yt-dlp/lib/python3.9/site-packages/yt_dlp/YoutubeDL.py", line 1349, in __extract_info
    return self.process_ie_result(ie_result, download, extra_info)
  File "/Users/scott/.local/pipx/venvs/yt-dlp/lib/python3.9/site-packages/yt_dlp/YoutubeDL.py", line 1399, in process_ie_result
    ie_result = self.process_video_result(ie_result, download=download)
  File "/Users/scott/.local/pipx/venvs/yt-dlp/lib/python3.9/site-packages/yt_dlp/YoutubeDL.py", line 2430, in process_video_result
    self.process_info(new_info)
  File "/Users/scott/.local/pipx/venvs/yt-dlp/lib/python3.9/site-packages/yt_dlp/YoutubeDL.py", line 2836, in process_info
    partial_success, real_download = self.dl(fname, new_info)
  File "/Users/scott/.local/pipx/venvs/yt-dlp/lib/python3.9/site-packages/yt_dlp/YoutubeDL.py", line 2575, in dl
    new_info = copy.deepcopy(self._copy_infodict(info))
  File "/Users/scott/.pyenv/versions/3.9.7/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/copy.py", line 146, in deepcopy
    y = copier(x, memo)
  File "/Users/scott/.pyenv/versions/3.9.7/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/copy.py", line 230, in _deepcopy_dict
    y[deepcopy(key, memo)] = deepcopy(value, memo)
  File "/Users/scott/.pyenv/versions/3.9.7/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/copy.py", line 172, in deepcopy
    y = _reconstruct(x, memo, *rv)
  File "/Users/scott/.pyenv/versions/3.9.7/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/copy.py", line 270, in _reconstruct
    state = deepcopy(state, memo)
  File "/Users/scott/.pyenv/versions/3.9.7/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/copy.py", line 146, in deepcopy
    y = copier(x, memo)
  File "/Users/scott/.pyenv/versions/3.9.7/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/copy.py", line 230, in _deepcopy_dict
    y[deepcopy(key, memo)] = deepcopy(value, memo)
  File "/Users/scott/.pyenv/versions/3.9.7/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/copy.py", line 161, in deepcopy
    rv = reductor(4)
TypeError: cannot pickle 'generator' object
```

**Comment by pukkandan** [comment]
Fixed in 282f570918f936a3aa9f57d4c85de4693da882c9

**Comment by scolby33** [comment]
Thank you! I thought I tried master, but I must've done the `pip install git+https://...` incorrectly.
